### PR TITLE
001 job role api

### PR DESCRIPTION
**Pull request by ldevlin559**

### CODE REVIEW

File: README.md
Comment (mlynch147) [inline README.md:1]: Revert all changes in README.md - this will be in Amara's PR

File: src/main/java/org/example/daos/JobRoleDao.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class JobRoleDao {
+    public List<JobRole> getJobRoles() throws SQLException {
+        List<JobRole> jobRoles = new ArrayList<>();
+        try (Connection connection = DatabaseConnector.getConnection()) {
+            String query = "SELECT jobRoleId, roleName, location,"
+                    + " Capability.capabilityName, Band.bandName, "
+                    + "closingDate FROM `job-roles`"
+                    + " JOIN Capability using(capabilityId)"
+                    + " JOIN Band using(bandId)"
+                    + " WHERE closingDate > current_date();";
```
Comment (mlynch147) [inline src/main/java/org/example/daos/JobRoleDao.java:21]: wrong condition here - AC states that you must bring back jobs with a status of OPEN

File: src/main/java/org/example/daos/JobRoleDao.java
Comment (mlynch147) [inline src/main/java/org/example/daos/JobRoleDao.java:1]: No test class provided for this

File: src/main/java/org/example/mappers/JobRoleMapper.java
Comment (mlynch147) [inline src/main/java/org/example/mappers/JobRoleMapper.java:1]: No test class provided for this

File: .github/workflows/build.yml
Comment (ldevlin559) [inline .github/workflows/build.yml:1]: test

File: README.md
Comment (mlynch147) [reply README.md:1]: done

**Review by TaylorB-softdevkainos — APPROVED** [review]

File: src/test/java/org/example/mapper/JobRoleMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class JobRoleMapperTest {
+
+    JobRoleDao jobRoleDao = Mockito.mock(JobRoleDao.class);
```
Comment (mlynch147) [inline src/test/java/org/example/mapper/JobRoleMapperTest.java]: JobRoleMapper doesnt have any dependencies so you dont have to Mock the JobRoleDao

The JobRoleMapper is the actual class you are testing so it shouldnt be Mocked... this should be the real class...

and one final thing, ensure that the list has at least one item in it when testing....

**Review by mlynch147 — APPROVED** [review]

**Review by mlynch147 — APPROVED** [review]

**Pull request merged**